SimPEG/Directives.py

```diff
@@ -303,26 +303,27 @@ def endIter(self):
         # Set the weighting matrix to None so that it is recomputed next time
         # it is called in the inversion
         self.reg._W = None
+        self.reg._Wsmooth = None
```

Comment:
this is because I did not look at the 2Deriv... it was still looking to Wsmooth (which it should no longer be doing). I have pushed a change where we now compute the 2Deriv for all of the regularization components. This whole implementation will be cleaner when we break it apart further...